Jolt/Math/UVec4.h

```diff
@@ -212,7 +212,7 @@ class [[nodiscard]] alignas(JPH_VECTOR_ALIGNMENT) UVec4
 	};
 
 private:
-	static const UVec4			sFourMinusXShuffle[];
+	JPH_EXPORT static const UVec4	sFourMinusXShuffle[];
```

Comment:
This means that a call to ShiftComponents4Minus will have to fetch the pointer to sFourMinusXShuffle and then dereference it to fetch the actual address of the sFourMinusXShuffle table, whereas before it would be able to load the constant address of the sFourMinusXShuffle table directly. This is problematic as the intrinsics are usually used in tight inner loops.

Reply:
Since it's only used in UVec4::ShiftComponents4Minus, would you agree to just put the table into the inline function instead?

Reply:
Regarding the pointer fetch: are you certain about that or is it an assumption? Currently when I leave out the JPH_EXPORT, only the UnitTests project fails to link, the others (e.g. the Samples) work fine. That's because the UnitTests call those inline functions and therefore need to link to the table.
My point is, that pointer fetch should only be necessary for external code. Everything internal to the Jolt library should still be able to load the data directly (though I haven't looked at any generated assembly). Since all the performance critical code is in Jolt, it shouldn't make a difference and the export is only necessary so that the unit tests still compile (which IMO aren't that performance critical).

Reply:
I looked at the disassembly yesterday, and it is indeed only an issue outside of the Jolt dll. Inside the Jolt dll it uses a ```lea register, constant address``` while outside it uses a ```mov register, address to pointer```.

I tried putting it inside UVec4::ShiftComponents4Minus as a local static but that gives horrible static initialization code using a thread local:

```
00007FF69B811A6B  mov         ecx,4  
00007FF69B811A70  mov         rax,qword ptr gs:[58h]  
00007FF69B811A79  mov         r8,qword ptr [rax]  
00007FF69B811A7C  mov         eax,dword ptr [rcx+r8]  
00007FF69B811A80  cmp         dword ptr [$TSS0 (07FF69BA49E40h)],eax  
00007FF69B811A86  jle         07FF69B811AD8h  
00007FF69B811A88  lea         rcx,[$TSS0 (07FF69BA49E40h)]  
00007FF69B811A8F  call        _Init_thread_header (07FF69B956EA8h)  
00007FF69B811A94  cmp         dword ptr [$TSS0 (07FF69BA49E40h)],0FFFFFFFFh  
00007FF69B811A9B  jne         07FF69B811AD8h  
00007FF69B811A9D  vpcmpeqd    xmm0,xmm0,xmm0  
00007FF69B811AA1  vmovdqu     xmmword ptr [sFourMinusXShuffle (07FF69BA49E50h)],xmm0  
00007FF69B811AA9  vmovaps     ymm1,ymmword ptr [__ymm@ffffffffffffffff0f0e0d0c0b0a0908ffffffffffffffffffffffff0f0e0d0c (07FF69B9FC660h)]  
00007FF69B811AB1  vmovups     ymmword ptr [sFourMinusXShuffle+10h (07FF69BA49E60h)],ymm1  
00007FF69B811AB9  vmovaps     ymm1,ymmword ptr [__ymm@0f0e0d0c0b0a09080706050403020100ffffffff0f0e0d0c0b0a090807060504 (07FF69B9FA8C0h)]  
00007FF69B811AC1  vmovups     ymmword ptr [sFourMinusXShuffle+30h (07FF69BA49E80h)],ymm1  
00007FF69B811AC9  lea         rcx,[$TSS0 (07FF69BA49E40h)]  
00007FF69B811AD0  vzeroupper  
00007FF69B811AD3  call        _Init_thread_footer (07FF69B956E48h)  
00007FF69B811AD8  vpshufb     xmm0,xmm6,xmmword ptr [sFourMinusXShuffle+20h (07FF69BA49E70h)]  
```

Using ```static constexpr UVec4 sFourMinusXShuffle[] = ...``` inside the function declaration doesn't work either since the intrinsics are not constexpr, so I went with ```static constexpr uint32 sFourMinusXShuffle[][]``` aligned to 16 bytes and that seems to work.